src/java.base/share/classes/java/lang/ClassLoader.java

```diff
@@ -682,7 +682,7 @@ protected Object getClassLoadingLock(String className) {
     private void checkPackageAccess(Class<?> cls, ProtectionDomain pd) {
         final SecurityManager sm = System.getSecurityManager();
         if (sm != null) {
-            if (ReflectUtil.isNonPublicProxyClass(cls)) {
+            if (ReflectUtil.isNonPublicProxyClass(cls) || ReflectUtil.isMethodHandleProxiesClass(cls)) {
```

Comment:
Why do you need this?   `Proxy` allows non-public interface whereas `MethodHandleProxies` only allows public interface.